throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+//ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
+}
```

Comment:
Frances Campoy has a similar pkg here https://talks.golang.org/2014/go4java/runner/runner.go but without context (hey it was 2014)